### PR TITLE
created a class that checks if a users email is on the db and then re…

### PR DESCRIPTION
**Pull request by gidspektor**
…turns their id and if they arent on the db then it puts them on the db

### CODE REVIEW

File: login.php [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 $db = new PDO('mysql:host=127.0.0.1;dbname=users','root');
```
Comment (mattap86) [inline login.php]: maybe put in a function and place in other file to add to .gitignore and call $db = dbconnect func. here instead?

File: src/Users.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
+<?php
+
```
Comment (mattap86) [inline src/Users.php]: unnecessary lb

File: src/Users.php [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+namespace theNamespace;
+
+
+class Users
+{
+    public $DbConnection;
+    public $userId;
+    public $email;
+
+    /*
```
Comment (mattap86) [inline src/Users.php]: add * to first comment line for uniformity with rest of cadebase

File: src/Users.php [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+     * constructor that requires a db connection and an email and then puts them into variables
+     *
+     * @param $DbConnection is a db connection
+     *
```
Comment (mattap86) [inline src/Users.php]: can remove lb between params

File: src/Users.php [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+class Users
+{
+    public $DbConnection;
+    public $userId;
+    public $email;
+
+    /*
+     * constructor that requires a db connection and an email and then puts them into variables
+     *
+     * @param $DbConnection is a db connection
```
Comment (mattap86) [inline src/Users.php]: add datatype of param (pdo)

File: src/Users.php [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+     *
+     * @param $email is the entered email
```
Comment (mattap86) [inline src/Users.php]: add datatype of email (string?)

File: src/Users.php [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+     */
+    public function __construct($DbConnection, $email)
+    {
+        $this->DbConnection = $DbConnection;
+        $this->email = $email;
+    }
+
+    /*
```
Comment (mattap86) [inline src/Users.php]: add * to first comment line for uniformity

File: src/Users.php [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+     * this grabs the users ID from the db and returns it into an array
+     *
+     *
+     *@return this sets the userId param to contain an array containing the users ID
```
Comment (mattap86) [inline src/Users.php]: state variable returned and it's datatype in return

File: src/Users.php [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+     */
+    public function __construct($DbConnection, $email)
+    {
+        $this->DbConnection = $DbConnection;
+        $this->email = $email;
+    }
+
+    /*
+     * this grabs the users ID from the db and returns it into an array
+     *
```
Comment (mattap86) [inline src/Users.php]: unnecessary lb

File: src/Users.php [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+     *
+     *@return this sets the userId param to contain an array containing the users ID
+     */
+    public function grabIdFromDb()
```
Comment (mattap86) [inline src/Users.php]: type hint return?

File: src/Users.php [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+    {
+        $email = $this->email;
+        $query=$this->DbConnection->prepare("SELECT `id` FROM `users` WHERE `email` = :email;");
+        $query->bindParam(':email',$email);
+        $query->execute();
+        $result = $query->fetchAll();
+        return $this->userId = $result;
+    }
+
+    /*
```
Comment (mattap86) [inline src/Users.php]: add * to first comment line for uniformity

File: src/Users.php [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+     * this inserts new emails into the db if they don't already exist in there
+     *
+     * @param $newEmail, this is the new email that the user has entered
```
Comment (mattap86) [inline src/Users.php]: add datatype of variable to param

File: src/Users.php [hunk 13]
```diff
@@ -0,0 +1,79 @@
...
+     *
+     * @return, this inserts the new email into the db
+     */
+    public function insertEmailToDb($newEmail)
```
Comment (mattap86) [inline src/Users.php]: type hint

File: src/Users.php [hunk 14]
```diff
@@ -0,0 +1,79 @@
...
+    {
+        $query=$this->DbConnection->prepare("INSERT INTO `users` (`email`) VALUES (:email);");
+        $query->bindParam(':email',$newEmail);
+        return $query->execute();
+    }
+
+    /*
```
Comment (mattap86) [inline src/Users.php]: add * to first comment line for uniformity

File: src/Users.php [hunk 15]
```diff
@@ -0,0 +1,79 @@
...
+     * this checks if the userId param contains an ID, if it does then the session starts.
+     * If it doesnt contain an ID then it means the email doesnt exist on the db
+     * and runs the function to insert it into the db then starts a session.
+     *
+     * @return runs a function to put email into db if not already and starts a session and goes to pokedex
```
Comment (mattap86) [inline src/Users.php]: nothing is returned in this function?

File: src/Classes/Users.php [hunk 16]
```diff
@@ -0,0 +1,68 @@
...
+    {
+        $this->DbConnection = $DbConnection;
+        $this->email = $email;
+    }
+
+    /**
+     * this grabs the users ID from the db if it exists and sets the objects $userId with that value
+     * If there is no id matching the email then then it does not set the objects id.
+     */
+    public function grabIdFromDb()
```
Comment (jaholt) [inline src/Classes/Users.php]: does this need a return type hint?

**Pull request merged**